adapters/adnuntius/adnuntius_utils.go

```diff
@@ -0,0 +1,202 @@
+package adnuntius
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+)
+
+func setHeaders(ortbRequest openrtb2.BidRequest) http.Header {
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	if ortbRequest.Device != nil {
+		if ortbRequest.Device.IP != "" {
+			headers.Add("X-Forwarded-For", ortbRequest.Device.IP)
+		}
+		if ortbRequest.Device.UA != "" {
+			headers.Add("user-agent", ortbRequest.Device.UA)
+		}
+	}
+	return headers
+}
+
+func makeEndpointUrl(ortbRequest openrtb2.BidRequest, a *adapter, noCookies bool) (string, []error) {
+	uri, err := url.Parse(a.endpoint)
+	endpointUrl := a.endpoint
```

Comment:
The url.Parse function is called but the returned `uri` is not used correctly. The endpointUrl should be constructed using uri.String() instead of a.endpoint.